### ISSUE TITLE
Nonfunctioning semantic announcements on Ubuntu desktop

### ISSUE THREAD

**Issue by ashuntu** [flutter/flutter#172023]
### Steps to reproduce

Flutter on Ubuntu Linux desktop doesn't seem to support semantic announcements or live regions. `SemanticsService.isAnnounceSupported()` outputs `true`.

1. Create the default new Flutter application via VSCode
2. Add a `SemanticsService.announce` to the `_incrementCounter` function
3. Enable the screen reader on Ubuntu desktop
4. Run the app
5. Press the + button to increment the counter
6. Observe the screen reader not announcing anything

In my example below, I've also added a live region and the output of `SemanticsService.isAnnounceSupported()`.

I have no special settings for the screen reader or Ubuntu, everything should be near defaults.

### Expected results

The screen reader should read aloud when a live region changes content, or when a semantics announcement is manually triggered.

### Actual results

No announcements are read by the screen reader.

### Code sample

<details open><summary>Code sample</summary>

Slightly modified version of the default new application via VSCode:

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });

    SemanticsService.announce(
      'Incrementing counter',
      TextDirection.ltr,
    ); // doesn't actually say anything
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('You have pushed the button this many times:'),
            Semantics(
              liveRegion:
                  true, // doesn't actually result in any expected behavior
              child: Text(
                '$_counter',
                style: Theme.of(context).textTheme.headlineMedium,
              ),
            ),
            Text(
              'Supports announce: ${SemanticsService.isAnnounceSupported()}', // outputs true
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

</details>

### Logs

[Logs (txt file)](https://github.com/user-attachments/files/21188715/logs.txt)

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
$ flutter doctor -v
[✓] Flutter (Channel stable, 3.32.5, on Ubuntu 24.04.2 LTS 6.11.0-29-generic, locale en_US.UTF-8) [27ms]
    • Flutter version 3.32.5 on channel stable at /home/[USERNAME]/fvm/versions/3.32.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fcf2c11572 (2 weeks ago), 2025-06-24 11:44:07 -0700
    • Engine revision dd93de6fb1
    • Dart version 3.8.1
    • DevTools version 2.45.1

[✗] Android toolchain - develop for Android devices [33ms]
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/linux-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome) [12ms]
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop [117ms]
    • Ubuntu clang version 18.1.3 (1ubuntu1)
    • cmake version 3.28.3
    • ninja version 1.11.1
    • pkg-config version 1.8.1
    ! Unable to access driver information using 'eglinfo'.
      It is likely available from your distribution (e.g.: apt install mesa-utils)

[!] Android Studio (not installed) [10ms]
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/linux-android-setup for detailed instructions).

[✓] VS Code (version unknown) [10ms]
    • VS Code at /snap/code/current/usr/share/code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
    ✗ Unable to determine VS Code version.

[✓] Connected device (1 available) [37ms]
    • Linux (desktop) • linux • linux-x64 • Ubuntu 24.04.2 LTS 6.11.0-29-generic

[✓] Network resources [178ms]
    • All expected network resources are available.

! Doctor found issues in 3 categories.
```

</details>

**Comment by danagbemava-nc** [comment]
Reproducible using the code sample provided above. 

On master, `SemanticsService.isAnnounceSupported()` has been removed but that doesn't really have any effect on the announcement not working. 

<details>
<summary>flutter doctor -v</summary>

```
[!] Flutter (Channel master, 3.33.0-1.0.pre.947, on Ubuntu 25.04 6.14.0-15-generic, locale en_US.UTF-8) [3.4s]
    • Flutter version 3.33.0-1.0.pre.947 on channel master at /home/daniel/fvm/versions/master
    ! Warning: `flutter` on your path resolves to /home/daniel/fvm/versions/stable/bin/flutter, which is not inside your current Flutter SDK checkout at /home/daniel/fvm/versions/master. Consider adding /home/daniel/fvm/versions/master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /home/linuxbrew/.linuxbrew/Cellar/dart/3.2.2/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /home/daniel/fvm/versions/master. Consider adding /home/daniel/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 63f471185f (21 hours ago), 2025-07-13 08:12:37 -0400
    • Engine revision 63f471185f
    • Dart version 3.9.0 (build [3.9.0-333.0.dev](http://3.9.0-333.0.dev/))
    • DevTools version 2.48.0
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, enable-swift-package-manager
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0) [8.5s]
    • Android SDK at /home/daniel/Android/Sdk
    • Emulator version 33.1.23.0 (build_id 11150993) (CL:N/A)
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/daniel/Android/Sdk
    • Java binary at: /home/daniel/.local/share/JetBrains/Toolbox/apps/android-studio/jbr/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.
[✓] Chrome - develop for the web [59ms]
    • Chrome at google-chrome
[✓] Linux toolchain - develop for Linux desktop [1,654ms]
    • Ubuntu clang version 20.1.2 (0ubuntu1)
    • cmake version 3.31.6
    • ninja version 1.12.1
    • pkg-config version 1.8.1
    • OpenGL core renderer: Mesa Intel(R) UHD Graphics (CML GT2) (X11)
    • OpenGL core version: 4.6 (Core Profile) Mesa 25.0.3-1ubuntu2 (X11)
    • OpenGL core shading language version: 4.60 (X11)
    • OpenGL ES renderer: Mesa Intel(R) UHD Graphics (CML GT2) (X11)
    • OpenGL ES version: OpenGL ES 3.2 Mesa 25.0.3-1ubuntu2 (X11)
    • OpenGL ES shading language version: OpenGL ES GLSL ES 3.20 (X11)
    • GL_EXT_framebuffer_blit: yes (X11)
    • GL_EXT_texture_format_BGRA8888: yes (X11)
[✓] Android Studio (version 2023.1) [45ms]
    • Android Studio at /home/daniel/.local/share/JetBrains/Toolbox/apps/android-studio
    • Flutter plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
[✓] VS Code (version 1.91.1) [36ms]
    • VS Code at /usr/share/code
    • Flutter extension version 3.110.0
[✓] Connected device (2 available) [543ms]
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 25.04 6.14.0-15-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 126.0.6478.182
[✓] Network resources [1,500ms]
    • All expected network resources are available.
! Doctor found issues in 1 category.
```

```
[!] Flutter (Channel stable, 3.32.6, on Ubuntu 25.04 6.14.0-15-generic, locale en_US.UTF-8) [453ms]
    • Flutter version 3.32.6 on channel stable at /home/daniel/fvm/versions/stable
    ! Warning: `dart` on your path resolves to /home/linuxbrew/.linuxbrew/Cellar/dart/3.2.2/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /home/daniel/fvm/versions/stable. Consider adding /home/daniel/fvm/versions/stable/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 077b4a4ce1 (6 days ago), 2025-07-08 13:31:08 -0700
    • Engine revision 72f2b18bb0
    • Dart version 3.8.1
    • DevTools version 2.45.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0) [6.7s]
    • Android SDK at /home/daniel/Android/Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /home/daniel/Android/Sdk
    • Java binary at: /home/daniel/.local/share/JetBrains/Toolbox/apps/android-studio/jbr/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.
[✓] Chrome - develop for the web [74ms]
    • Chrome at google-chrome
[✓] Linux toolchain - develop for Linux desktop [1,364ms]
    • Ubuntu clang version 20.1.2 (0ubuntu1)
    • cmake version 3.31.6
    • ninja version 1.12.1
    • pkg-config version 1.8.1
    • OpenGL core renderer: Mesa Intel(R) UHD Graphics (CML GT2) (X11)
    • OpenGL core version: 4.6 (Core Profile) Mesa 25.0.3-1ubuntu2 (X11)
    • OpenGL core shading language version: 4.60 (X11)
    • OpenGL ES renderer: Mesa Intel(R) UHD Graphics (CML GT2) (X11)
    • OpenGL ES version: OpenGL ES 3.2 Mesa 25.0.3-1ubuntu2 (X11)
    • OpenGL ES shading language version: OpenGL ES GLSL ES 3.20 (X11)
    • GL_EXT_framebuffer_blit: yes (X11)
    • GL_EXT_texture_format_BGRA8888: yes (X11)
[✓] Android Studio (version 2023.1) [64ms]
    • Android Studio at /home/daniel/.local/share/JetBrains/Toolbox/apps/android-studio
    • Flutter plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
[✓] VS Code (version unknown) [57ms]
    • VS Code at /usr/share/code
    • Flutter extension version 3.110.0
    ✗ Unable to determine VS Code version.
[✓] Connected device (2 available) [680ms]
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 25.04 6.14.0-15-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 126.0.6478.182
[✓] Network resources [974ms]
    • All expected network resources are available.
! Doctor found issues in 1 category.
```
</details>

**Comment by robert-ancell** [comment]
According to the [docs](https://api.flutter.dev/flutter/semantics/SemanticsService-class.html) the new APIs are:

```dart
SemanticsService.sendAnnouncement(
  View.of(context),
  'Incrementing counter',
  TextDirection.ltr,
);
```

```dart
'Supports announce: ${MediaQuery.supportsAnnounceOf(context)}'
```

Still shows true but doesn't announce.
